diff -Naur openv2g-src.orig/src/codec/EncoderChannel.c openv2g-src/src/codec/EncoderChannel.c
--- openv2g-src.orig/src/codec/EncoderChannel.c	2019-01-23 09:10:52.000000000 +0000
+++ openv2g-src/src/codec/EncoderChannel.c	2022-04-07 04:13:01.558549073 +0000
@@ -231,12 +231,14 @@
  * store the integer's value.
  */
 int encodeUnsignedIntegerBig(bitstream_t* stream, size_t size, uint8_t* data, size_t len) {
+#define _MAX_BIGINT_ARRAY 25
+
 	int errn = 0;
 	int i;
 	int lenM1 = len - 1;
-	const int MAX_BIGINT_ARRAY = 25;
+	const int MAX_BIGINT_ARRAY = _MAX_BIGINT_ARRAY;
 	uint8_t lastEncode = 0;
-	uint8_t bytesToShift[MAX_BIGINT_ARRAY]; // MAXIMUM
+	uint8_t bytesToShift[_MAX_BIGINT_ARRAY]; // MAXIMUM
 	size_t bitsToEncode = len * 8;
 
 	if(MAX_BIGINT_ARRAY <= len) {
