diff -Naur openv2g-src.orig/src/din/dinEXIDatatypes.h openv2g-src/src/din/dinEXIDatatypes.h
--- openv2g-src.orig/src/din/dinEXIDatatypes.h	2019-07-08 15:32:08.000000000 +0000
+++ openv2g-src/src/din/dinEXIDatatypes.h	2022-04-05 03:38:03.988431502 +0000
@@ -44,8 +44,12 @@
 
 #define SUPPORT_YES 1
 #define SUPPORT_NO 2
+#ifndef DEPLOY_DIN_CODEC
 #define DEPLOY_DIN_CODEC SUPPORT_NO
+#endif
+#ifndef DEPLOY_DIN_CODEC_FRAGMENT
 #define DEPLOY_DIN_CODEC_FRAGMENT SUPPORT_NO
+#endif
 
 #if DEPLOY_DIN_CODEC == SUPPORT_YES
 
diff -Naur openv2g-src.orig/src/iso1/iso1EXIDatatypes.h openv2g-src/src/iso1/iso1EXIDatatypes.h
--- openv2g-src.orig/src/iso1/iso1EXIDatatypes.h	2019-07-08 15:21:10.000000000 +0000
+++ openv2g-src/src/iso1/iso1EXIDatatypes.h	2022-04-05 03:44:55.952448808 +0000
@@ -44,8 +44,12 @@
 
 #define SUPPORT_YES 1
 #define SUPPORT_NO 2
+#ifndef DEPLOY_ISO1_CODEC
 #define DEPLOY_ISO1_CODEC SUPPORT_YES
+#endif
+#ifndef DEPLOY_ISO1_CODEC_FRAGMENT
 #define DEPLOY_ISO1_CODEC_FRAGMENT SUPPORT_YES
+#endif
 
 #if DEPLOY_ISO1_CODEC == SUPPORT_YES
 
diff -Naur openv2g-src.orig/src/iso2/iso2EXIDatatypes.h openv2g-src/src/iso2/iso2EXIDatatypes.h
--- openv2g-src.orig/src/iso2/iso2EXIDatatypes.h	2019-07-08 15:46:12.000000000 +0000
+++ openv2g-src/src/iso2/iso2EXIDatatypes.h	2022-04-05 03:45:37.728450563 +0000
@@ -44,8 +44,12 @@
 
 #define SUPPORT_YES 1
 #define SUPPORT_NO 2
+#ifndef DEPLOY_ISO2_CODEC
 #define DEPLOY_ISO2_CODEC SUPPORT_NO
+#endif
+#ifndef DEPLOY_ISO2_CODEC_FRAGMENT
 #define DEPLOY_ISO2_CODEC_FRAGMENT SUPPORT_NO
+#endif
 
 #if DEPLOY_ISO2_CODEC == SUPPORT_YES
 
diff -Naur openv2g-src.orig/src/xmldsig/xmldsigEXIDatatypes.h openv2g-src/src/xmldsig/xmldsigEXIDatatypes.h
--- openv2g-src.orig/src/xmldsig/xmldsigEXIDatatypes.h	2018-01-30 14:19:52.000000000 +0000
+++ openv2g-src/src/xmldsig/xmldsigEXIDatatypes.h	2022-04-05 03:46:10.824451953 +0000
@@ -44,8 +44,12 @@
 
 #define SUPPORT_YES 1
 #define SUPPORT_NO 2
+#ifndef DEPLOY_XMLDSIG_CODEC
 #define DEPLOY_XMLDSIG_CODEC SUPPORT_NO
+#endif
+#ifndef DEPLOY_XMLDSIG_CODEC_FRAGMENT
 #define DEPLOY_XMLDSIG_CODEC_FRAGMENT SUPPORT_NO
+#endif
 
 #if DEPLOY_XMLDSIG_CODEC == SUPPORT_YES
 
