diff --git a/CMakeLists.txt b/CMakeLists.txt
index bb930271e7..d783700132 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1614,6 +1614,7 @@ if(ENABLE_PLUGINS)
 		plugins/epan/stats_tree
 		plugins/epan/transum
 		plugins/epan/unistim
+		plugins/epan/v2g
 		plugins/epan/wimax
 		plugins/epan/wimaxasncp
 		plugins/epan/wimaxmacphy
diff --git a/packaging/nsis/wireshark.nsi b/packaging/nsis/wireshark.nsi
index 2a3e7c6183..1b322ca596 100644
--- a/packaging/nsis/wireshark.nsi
+++ b/packaging/nsis/wireshark.nsi
@@ -1029,6 +1029,8 @@ File "${STAGING_DIR}\plugins\${VERSION_MAJOR}.${VERSION_MINOR}\epan\irda.dll"
 File "${STAGING_DIR}\plugins\${VERSION_MAJOR}.${VERSION_MINOR}\epan\opcua.dll"
 File "${STAGING_DIR}\plugins\${VERSION_MAJOR}.${VERSION_MINOR}\epan\profinet.dll"
 File "${STAGING_DIR}\plugins\${VERSION_MAJOR}.${VERSION_MINOR}\epan\unistim.dll"
+File "${STAGING_DIR}\plugins\${VERSION_MAJOR}.${VERSION_MINOR}\epan\v2gexi.dll"
+File "${TOP_SRC_DIR}\plugins\epan\v2g\dissector\v2g.lua"
 File "${STAGING_DIR}\plugins\${VERSION_MAJOR}.${VERSION_MINOR}\epan\wimax.dll"
 File "${STAGING_DIR}\plugins\${VERSION_MAJOR}.${VERSION_MINOR}\epan\wimaxasncp.dll"
 File "${STAGING_DIR}\plugins\${VERSION_MAJOR}.${VERSION_MINOR}\epan\wimaxmacphy.dll"
